### DISCUSSION TITLE
Multiple Detectors: CPU1, CPU2, etc vs only one CPU

### DISCUSSION THREAD

**Discussion by stamandr** [blakeblackshear/frigate#2157] [Ask A Question]
I have a RPI4 4GB with 5 Hikvision cameras ( width: 704,height: 480 ). I'm currently playing around with testing multiple detectors. 

  cpu1:
    type: cpu
    num_threads: 3
  cpu2:
    type: cpu
    num_threads: 3
  cpu3:
    type: cpu
    num_threads: 3
  cpu4:
    type: cpu
    num_threads: 3
  cpu5:
    type: cpu
    num_threads: 3

I know that CPU Detectors are not recommended, but with Coral USB's being not available, there isn't much that can be done.

I have noticed that when all 5 CPU's are used ( 1 for each camera ) that inference speeds are 170-300+.
When I change it to one CPU only inference speeds are about the same or even lower. Processor use is about the same with both.

Are there really performance increases by using multiple CPU's ? Is there a recommended setting for those of us who do not have a Coral USB on our RPI4's ?

**Comment by blakeblackshear** [comment]
It should help when multiple cameras are running detection in parallel. Each detector process is completely independent, so they each have their own inference_time. Have multiple running doesn't have any overhead. The idle ones don't consume much resources. You may want to limit it if you want to limit the CPU usage.

**Reply by stamandr** [reply]
I've been doing a bunch of testing. My results are not great with multiple CPU Detectors.  Could I have something wrong ?

DEBUG 0.9.4-26AE608
detector	inference speed	detection start	pid
cpu1		320.37		0		217
cpu2		255.6		0		219
cpu3		337.29		0		220
cpu4		299.97		0		223
cpu5		327.51		0		224
CPU Utilization is between 40-80+%. Mostly around 50%

One CPU
cpu1		160.34		0		218
CPU Utilization is between 28-80+%. Mostly around 35%

Camera height: 480 & width: 704
FPS 5

**Reply by blakeblackshear** [reply]
If the inference times start to rise, that means they are fighting for resources, so you may be better off with less.
